### PR TITLE
podmanv2 add core container commands

### PR DESCRIPTION
**Pull request by baude**
add core container commands for podmanv2: kill, pause, restart, rm, stop, unpause

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podmanV2/containers/kill.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: killCommand,
+		Parent:  containerCmd,
+	})
+	flags := killCommand.Flags()
+	flags.BoolVarP(&killOptions.All, "all", "a", false, "Signal all running containers")
+	flags.StringVarP(&signal, "signal", "s", "KILL", "Signal to send to the container")
+	flags.BoolVarP(&killOptions.Latest, "latest", "l", false, "Act on the latest container podman is aware of")
```
Comment (TomSweeneyRedHat) [inline cmd/podmanV2/containers/kill.go]: 'podman' -> 'Podman' through out except for command examples.

File: cmd/podmanV2/containers/restart.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/containers/libpod/libpod/define"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	restartDescription = `Restarts one or more running containers. The container ID or name can be used.
+
+  A timeout before forcibly stopping can be set, but defaults to 10 seconds.`
```
Comment (TomSweeneyRedHat) [inline cmd/podmanV2/containers/restart.go]: stopping or restarting?

Comment (TomSweeneyRedHat) [reply cmd/podmanV2/containers/restart.go]: "but defaults" -> "and defaults"  maybe

File: cmd/podmanV2/containers/restart.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: restartCommand,
+		Parent:  containerCmd,
+	})
+	flags := restartCommand.Flags()
+	flags.BoolVarP(&restartOptions.All, "all", "a", false, "Restart all non-running containers")
+	flags.BoolVarP(&restartOptions.Latest, "latest", "l", false, "Act on the latest container podman is aware of")
+	flags.BoolVar(&restartOptions.Running, "running", false, "Restart only running containers when --all is used")
+	flags.UintVarP(&restartTimeout, "timeout", "t", define.CtrRemoveTimeout, "Seconds to wait for stop before killing the container")
+	flags.UintVar(&restartTimeout, "time", define.CtrRemoveTimeout, "Seconds to wait for stop before killing the container")
```
Comment (TomSweeneyRedHat) [inline cmd/podmanV2/containers/restart.go]: Are time and timeout effectively equivalent?

File: cmd/podmanV2/containers/rm.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	rmDescription = fmt.Sprintf(`Removes one or more containers from the host. The container name or ID can be used.
+
+  Command does not remove images. Running or unusable containers will not be removed without the -f option.`)
```
Comment (TomSweeneyRedHat) [inline cmd/podmanV2/containers/rm.go]: unusable or usable?

**Comment by baude** [comment]
fwiw, all of the cli descriptions and flags are moved untouched.

File: cmd/podmanV2/root.go [hunk 5]
```diff
@@ -23,7 +23,8 @@ var rootCmd = &cobra.Command{
 func init() {
 	// Override default --help information of `--version` global flag}
 	var dummyVersion bool
-	rootCmd.PersistentFlags().BoolVarP(&dummyVersion, "version", "v", false, "Version of podman")
+	// TODO had to disable shorthand -v for version due to -v rm with volume
+	rootCmd.PersistentFlags().BoolVar(&dummyVersion, "version", false, "Version of podman")
```
Comment (TomSweeneyRedHat) [inline cmd/podmanV2/root.go]: Podman

File: cmd/podmanV2/containers/rm.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	)
+	// Storage conflicts with --all/--latest/--volumes/--cidfile/--ignore
+	if rmOptions.Storage {
+		if rmOptions.All || rmOptions.Ignore || rmOptions.Latest || rmOptions.Volumes || rmOptions.CIDFiles != nil {
+			return errors.Errorf("--storage conflicts with --volumes, --all, --latest, --ignore and --cidfile")
+		}
+	}
+	responses, err := registry.ContainerEngine().ContainerRm(context.Background(), args, rmOptions)
+	if err != nil {
+		// TODO exitcode is a global main variable to track exit codes.
```
Comment (mheon) [inline cmd/podmanV2/containers/rm.go]: Let's not merge until this is actually fixed - shouldn't be that bad, it looks like just a bit of code in main

File: pkg/signal/signal_linux.go [hunk 7]
```diff
@@ -125,3 +126,14 @@ func StopCatch(sigc chan os.Signal) {
 	signal.Stop(sigc)
 	close(sigc)
 }
+
+// SignalToString converts a given signal to its string
+// representation
+func SignalToString(sig syscall.Signal) (string, error) {
```
Comment (jwhonce) [inline pkg/signal/signal_linux.go]:
```suggestion
func ToString(sig syscall.Signal) (string, error) {
```

signal.ToString()  is better than the stutter of single.SignalToString()

File: cmd/podmanV2/containers/kill.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+var (
+	killOptions = entities.KillOptions{}
+	signal      string
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: killCommand,
+		Parent:  containerCmd,
```
[the remark this answers is not in the document]
Comment (baude) [reply cmd/podmanV2/containers/kill.go]: in your example, the subcommand 'list' does not have any flags.  so either it is incomplete or we have a flaw.  i'm not anticipating that you will duplicate all of those flags?

File: cmd/podmanV2/containers/rm.go (hunk 6 above)
Comment (baude) [reply cmd/podmanV2/containers/rm.go]: im not ok with either suggestion ... we should hash this out.  holding this comment for now.

**Comment by baude** [comment]
updated comments, i would prefer to merge and then talk about the exitcode business...

**Pull request merged**